pallene/coder.lua

```diff
@@ -1291,15 +1331,35 @@ gen_cmd["SetField"] = function(self, cmd, _func)
 end
 
 gen_cmd["NewClosure"] = function (self, cmd, _func)
+    local func = self.module.functions[cmd.f_id]
+
+    local upvals = {}
+    for i, upval_info in ipairs(func.captured_vars) do
+        local c_val = self:c_value(upval_info.value)
+        upvals[i] = { c_val = c_val, typ = upval_info.decl.typ }
+    end
+
+    local capture_upvalues = {}
+    for i, c_upval in ipairs(upvals) do
```

Comment:
Do we need the `upvals` local variable? Can we merge these two for loops into a single loop?